src/lib/data/operations/aggregation/feature/feature-aggregation.ts

```diff
@@ -0,0 +1,60 @@
+import { CartoError } from '../../../../core/errors/CartoError';
+import { AggregationType } from '../AggregationType';
+
+export interface AggregatedFeatureProperties {
+  aggregatedValue: number;
+  featureCount: number;
+}
+
+export function aggregateFeatures(
+  features: AggregatedFeatureProperties[],
+  aggregation: AggregationType = '' as AggregationType
+) {
+  const aggregationData = aggregation.split('_');
+  const aggregationName = aggregationData.shift();
+
+  const aggregationFunction =
+    aggregationFunctions[aggregationName?.toLowerCase() as AggregationType];
+
+  if (!aggregationFunction) {
+    throw new CartoError({
+      type: '[DataView]',
+      message: `"${aggregation}" feature aggregation type not implemented`
+    });
+  }
+
+  return {
+    result: aggregationFunction(features, aggregationData),
+    nullCount: 0
+  };
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const aggregationFunctions: Record<AggregationType, Function> = {
+  [AggregationType.COUNT](aggregatedFeatures: AggregatedFeatureProperties[]) {
+    return aggregatedFeatures.reduce((total, feature) => total + feature.featureCount, 0);
+  },
+
+  [AggregationType.AVG](aggregatedFeatures: AggregatedFeatureProperties[]) {
+    return aggregationFunctions.sum(aggregatedFeatures) / aggregatedFeatures.length;
```

Comment:
Shouldn't this average be ponderated by the count?

Reply:
Yes, I did that at the beginning but then I had a conflict about AVGs and forgot to change it back 😁